apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguage.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useCallback } from "react";
+import { Button, Flex, Icons } from "@ledgerhq/react-ui";
+import Text from "~/renderer/components/Text";
+import DeviceLanguageInstallation from "./DeviceLanguageInstallation";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { idsToLanguage, Language, DeviceInfo } from "@ledgerhq/types-live";
+import { useTranslation } from "react-i18next";
+import { withV3StyleProvider } from "~/renderer/styles/StyleProviderV3";
+
+type Props = {
+  // this makes sure that this component is only rendered if languageId is present in deviceInfo
```

Comment:
typescript wise ok, but giving the state of static typing in the app, could you return null in case it's missing?

Reply:
It is already handled in the page that uses this component.
Not sure if doing this check as a default as a defensive programming is the best approach.
We'd have to do it everywhere.

Reply:
This is equivalent to asking for a mandatory non-null languageId as a separate prop